### PR TITLE
Suggestions

### PR DESCRIPTION
**Pull request by ccorcos**
- Use camelCase instead of snake-case for files and variables.
- Add newlines at the end of the file (editor can do this on save)
- No .vscode folder in this repo. You can add it yourself and gitignore if you want.

I want to see this logic broken up into smaller pieces that are independently tested. 
I've mocked out some test files for you but feel free to add to them.
- parseImports test.
- exportIsType test.
- exportIsClass test.
- classIsUsedAsValue test.
- parseDependencies test.
	- write some more tests for parseDependencies as well.
	- make sure to test out with nested files as such as well.
- importCycles test

### CODE REVIEW

**Comment by ccorcos** [comment]
Hey @QuentinGruber I see you're still working on this. Just let me know when you think its ready for review 😄

**Comment by QuentinGruber** [comment]
Hi @ccorcos for me it's ready, I don't think I forgot anything 🤠

File: src/classIsUsedAsValue.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		const importFileSource:FileSource = {
+			filePath: parserImport.libraryName,
+			source: classSource,
+		}
+		if(await checkIfImportExistAtRuntime(importFileSource,parserImport,entryFileSource.source,sourceFileParsed)){
+			return true
+		}
+	}
+	return false
+}
```
Comment (ccorcos) [inline src/classIsUsedAsValue.test.ts:23]: This function should be used in by the import cycles code somewhere, right?

File: src/import-cycles.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	it("should have 2 files that have cycles", async function () {
+		assert.equal(importCycles.length, 2)
+	})
+	it("entry.ts should have 3 cycles", async function () {
+		assert.equal(importCycles[0].cycle.length, 3)
+	})
+	it("entry2.ts should have 1 cycles", async function () {
+		assert.equal(importCycles[1].cycle.length, 1)
+	})
+})
```
Comment (ccorcos) [inline src/import-cycles.test.ts]: These tests are really hard to follow. I have no idea what cycles we should be seeing or not...

I think this can be improved by:
1. write the source code for each file directly in this test. write them to a tmp folder to execute the test. 
2. Write some comments, etc., that explain what the intention is for each line in the files that we're creating. 
3. It looks like we have multiple assertions here that might be better broken up into more isolated test examples. What about a single entry point and a single cycle? Simpler tests would be better.

**Comment by ccorcos** [comment]
This is looking great @QuentinGruber. Just one more comment to take a look at.

**Comment by ccorcos** [comment]
Also can we format all this code using prettier. 
```
{
	"semi": false,
	"useTabs": true,
	"trailingComma": "es5"
}
```

File: src/import-cycles.test.ts (hunk 2 above)
Comment (QuentinGruber) [reply src/import-cycles.test.ts]: It's done feel free to let me know if you want me to do more test cases 🤠.

**Comment by ccorcos** [comment]
@QuentinGruber this looks great!
